### ISSUE TITLE
Interfaces Bug/Query

### ISSUE THREAD

**Issue by sukhysall** [mylar3/mylar3#287]
Question: Does the "Interface" option work?

I created a copy of the "default" folder and renamed it to "carbon". I then selected this new interface from the Advanced Settings options. For good measure I then did a restart. All good so far.

So....I then started making changes to the style.css to alter some of the colour schemes to more of a Dark Mode theme but I noticed no changes to the app after a page refresh and a restart for good measure.

When looking at at the page using the Firefox developer F12 option, I highlighted the styling sheet from which the page was pulling it's styling and the path led back to the default option rather then the carbon option. 

I was expecting that when selecting the "carbon" interface option, that all pages would then pull from the "carbon" folder. 

Please note: Whilst the carbon interface option was still the selected option. I went and made style.css change in the "interface" folder, did a page refresh and it worked successfully.

Example screenshot below.

Thanks

[Default Interfaces](https://user-images.githubusercontent.com/5501195/82129597-eda65180-97bb-11ea-86db-be445e7cd981.png)

**Comment by barbequesauce** [comment]
Right now it doesnt do anything. It's a leftover from the original Headphones framework that was adapted for use with comics. I can say with certainty that interface changes that I've made do not respect the config values that should point to the correct interface.... and I know I'm not the only one :) It's a major overhaul to adapt the interface... but if one is determined, one can achieve anything (one page at a time)

**Comment by evilhero** [comment]
Everything was hardcoded to be default within the function calls - everything would ha e to be modified to use the config variable which holds the theme name (assuming its named the same as the folder). It's not impossible, just abit of work to get all the refs changed over correctly.

**Comment by sukhysall** [comment]
Ok.....that sounds......intensive.

@evilhero this could be reconsidered as a feature request :)

HOWEVER, if you deem it unfeasible to add the interface feature, I have a question: would my manually changing the styling cause any potential issues when it came to updates?

I ask because the in-app update didn't work from 0.3.5 to 0.3.6 so I then tried a git pull which also failed due to the changes I made to the style.css

In the end I resolved by renaming my original Mylar folder and git cloning from scratch.

**Comment by barbequesauce** [comment]
The way that git works is to compare the filesystem on your machine to "origin" - the copy on github. If there are differences, it won't overwrite your copies.

That said, you could create a pull request - basically upload your recommended changes - and then (once we approve) they become part of origin and you can reset your install to match origin again...

**Comment by sukhysall** [comment]
@barbequesauce I will do just that. I suspect it may not work out so well especially if most users are happy with the default colour scheme as what I have done so far tries at best to emulate the slate bootstrap theme I use for lazy librarian (so effectively a dark theme which may not meet most people's tastes)

But when free next I'll push what I have done just so you guys can take a look. Would love to hear your feedback.

**Comment by barbequesauce** [comment]
Absolutely, happy to have the help!

For what it’s worth - Mylar is the only light interface I have, I’d welcome it :)

**Comment by barbequesauce** [comment]
I’ll create a branch for you called dark-theme - when you create the PR do it against that branch please...

**Comment by evilhero** [comment]
All for any kind of help when it comes to improving the GUI :) 

As for the hard-coded part, I looked into it a bit more. The only changes that are needed are in the .html files themselves. All of them (or a good majority) point the .css files to the  ```data/interfaces/default/css``` folder, so you have to manually change those to the new css directory location path. So in a given html you might find something like: 
```
<%def name="headIncludes()">
        <link rel="stylesheet" href="interfaces/default/css/data_table.css">
        <link rel="stylesheet" href="interfaces/default/css/style.css">
</%def>
```
In the ```carbon``` directory, wherein you have all the html files, you'd have to go into each one and just change the reference to:
```
<%def name="headIncludes()">
        <link rel="stylesheet" href="interfaces/carbon/css/data_table.css">
        <link rel="stylesheet" href="interfaces/carbon/css/style.css">
</%def>
```
That way when the interface changes the css files are pointed to the correct location, and you'll see the changes.

**Comment by sukhysall** [comment]
Hey guys

Question. So i cloned via the command:

git clone https://github.com/mylar3/mylar3.git --branch dark-theme mylar

And install from dark-theme branch has been fine.

However, having made a few changes so far (still need to do the data_table.css) I'm finding an error when trying to git push:

"remote: Permission to mylar3/mylar3.git denied to sukhysall.
fatal: unable to access 'https://github.com/mylar3/mylar3/': The requested URL returned error: 403"

Have I missed a step when initially doing the git clone?

**Comment by barbequesauce** [comment]
Initial issue solved; new questions reasked in #290
